packages/eslint-plugin/src/rules/lines-around-comment.ts

```diff
@@ -0,0 +1,528 @@
+  });
+  return lines;
+}
+
+export default util.createRule<Options, MessageIds>({
+  name: 'lines-around-comment',
+  meta: {
+    type: 'layout',
+    docs: {
+      description: 'Require empty lines around comments',
+      recommended: false,
+      extendsBaseRule: true,
+    },
+    schema: {
+      type: 'array',
+      items: [
+        {
+          type: 'object',
+          properties: {
+            beforeBlockComment: {
+              type: 'boolean',
+              default: true,
+            },
+            afterBlockComment: {
+              type: 'boolean',
+              default: false,
+            },
+            beforeLineComment: {
+              type: 'boolean',
+              default: false,
+            },
+            afterLineComment: {
+              type: 'boolean',
+              default: false,
+            },
+            allowBlockStart: {
+              type: 'boolean',
+              default: false,
+            },
+            allowBlockEnd: {
+              type: 'boolean',
+              default: false,
+            },
+            allowClassStart: {
+              type: 'boolean',
+            },
+            allowClassEnd: {
+              type: 'boolean',
+            },
+            allowObjectStart: {
+              type: 'boolean',
+            },
+            allowObjectEnd: {
+              type: 'boolean',
+            },
+            allowArrayStart: {
+              type: 'boolean',
+            },
+            allowArrayEnd: {
+              type: 'boolean',
+            },
+            allowInterfaceStart: {
+              type: 'boolean',
+            },
+            allowInterfaceEnd: {
+              type: 'boolean',
+            },
+            allowTypeStart: {
+              type: 'boolean',
+            },
+            allowTypeEnd: {
+              type: 'boolean',
+            },
+            ignorePattern: {
+              type: 'string',
+            },
+            applyDefaultIgnorePatterns: {
+              type: 'boolean',
+            },
+          },
+          additionalProperties: false,
+        },
+      ],
+    },
+    fixable: baseRule.meta.fixable,
+    hasSuggestions: baseRule.meta.hasSuggestions,
+    messages: baseRule.meta.messages,
+  },
+  defaultOptions: [
+    {
+      beforeBlockComment: true,
+    },
+  ],
+  create(context, [_options]) {
+    const options = _options!;
+    const defaultIgnoreRegExp = COMMENTS_IGNORE_PATTERN;
+    const customIgnoreRegExp = new RegExp(options.ignorePattern ?? '', 'u');
+
+    const sourceCode = context.getSourceCode();
+    const comments = sourceCode.getAllComments();
+
+    const lines = sourceCode.lines;
+    const numLines = lines.length + 1;
+    const commentLines = getCommentLineNums(comments);
+    const emptyLines = getEmptyLineNums(lines);
+    const commentAndEmptyLines = new Set(commentLines.concat(emptyLines));
+
+    /**
+     * Returns whether or not comments are on lines starting with or ending with code
```

Comment:
Nit: I realize these JSDocs are mostly copypasta, but I think we* have a more ... refined JSDoc standard here 😄 

```suggestion
     * @returns Whether comments are on lines starting with or ending with code.
```

*We = me. I don't think we have a standard in the repo... but, this is a more succinct comment and plays better with intellisense.